helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -1135,6 +1145,34 @@ private void removeMessageFromZK(HelixDataAccessor accessor, Message message,
     }
   }
 
+  private void startMessageRefreshThread() {
+    int periodicMessageRefreshInterval =
+        HelixUtil.getSystemPropertyAsInt(SystemPropertyKeys.PERIODIC_MESSAGE_REFRESH_INTERVAL, -1);
```

Comment:
Another point is that:
 OnMessage called by the timer thread can throw exception. The thing is that I suspect that timer's internal thread may not schedule more OnMessage() if the previous one has exception thrown. This needs to be tested and verified. Otherwise, the timer thread may never work after exception thrown if my suspicion is right.

Java Concurrency Control book talked in detail about this issue. I suggest let at least let test throwing exception from OnMessage and see if this is a problem.

Reply:
Resolving this comment since we do not directly call onMessage in the current design.